framework/sessions/__init__.py

```diff
@@ -44,6 +47,26 @@ def add_key_to_url(url, scheme, key):
     parsed_redirect_url = parsed_url._replace(**replacements)
     return urlparse.urlunparse(parsed_redirect_url)
 
+def get_node_id(url):
+    try:
+        node_id = [x for x in url.split("/") if x][2]
```

Comment:
This isn't guaranteed to work, as we support the older `/project/<guid>/` url form, as well as just `/<guid>/`.